app/questionnaire/questionnaire_store_updater.py

```diff
@@ -496,7 +496,11 @@ def remove_dependent_blocks_and_capture_dependent_sections(self) -> None:
             section_key,
             blocks_to_remove,
         ) in self.dependent_block_id_by_section_key.items():
-            if section_key not in self.started_section_keys():
+            if section_key not in self.started_section_keys() and not any(
```

Comment:
Maybe worth adding a comment here to explain why this additional check has been added?